source/common/thread_local/thread_local_impl.cc

```diff
@@ -92,14 +92,23 @@ Event::PostCb InstanceImpl::SlotImpl::dataCallback(const UpdateCb& cb) {
   };
 }
 
-void InstanceImpl::SlotImpl::runOnAllThreads(const UpdateCb& cb, Event::PostCb complete_cb) {
+void InstanceImpl::SlotImpl::runOnAllThreads(const UpdateCb& cb, const Event::PostCb& complete_cb) {
```

Comment:
Do we still need the ASSERT stuff above? Can that be removed now? Did you end up finding a call site that does actually store a new value? Is that why we have these new variants?

If we need the new variants can you make sure to have unit tests that cover the ref counting stuff? I think I added them for all the existing variants.

Reply:
RE the ASSERTs I added in the previous PR: yes I think we still need them because I haven't removed the old API yet. I think maybe it's 1-2 more PRs before I can remove the old API or make it private in some way.

RE adding testcases for ref-counting (the alive-bit?) yeah. That makes sense; will do and then ping back.

Reply:
testcase added; ptal. I'm resolving this comment and I will remove the other asserts as soon as possible.